pkg/recipes/terraform/config/config_test.go

```diff
@@ -581,8 +608,10 @@ func Test_AddProviders(t *testing.T) {
 
 			tfconfig, err := New(ctx, testRecipeName, &envRecipe, &resourceRecipe, &tc.envConfig)
 			require.NoError(t, err)
-			for _, p := range tc.expectedUCPConfiguredProviders {
-				mProvider.EXPECT().BuildConfig(ctx, &tc.envConfig).Times(1).Return(p, nil)
+			if !tc.overrideUCPProviderConfig {
```

Comment:
It's really hard to understand that setting `overrideUCPProviderConfig` translates into building configs for providers configured on UCP. :) Can you share some details on how this flag should be used, or maybe make it more explicit?

Reply:
This check is to ensure we're building configs for UCP provider only when we're not overriding it. I added a comment.

Reply:
Can we call it useUCPProviderConfig then to make it more intuitive?